### PR TITLE
Rework tests wrt transient objects

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
This is the very first step towards having the tests run on actual TPMs.
In this step we
- Fix leaking integration tests
- Move TPM_startup into the simulator test harness
- Add a guard around integrations tests to look for transent objects before  and after the test.

### CODE REVIEW

**Comment by martinezjavier** [comment]
@AndreasFuchsSIT I've been told in the past that the tpm2-tss tests are not suitable to be run on real TPM devices due their destructive nature. And in fact here's a big scary warning in https://github.com/tpm2-software/tpm2-tss/blob/master/README.md#test-suite.

Could you please elaborate a little bit about this in the commit message? Are this warning only referring to the unit tests or also the integration tests?

**Comment by AndreasFuchsSIT** [comment]
@martinezjavier That warning is still totally true...
I'm starting the effort to enable running the testsuit on real TPMs.
I'll have to do a few more things, but then I'll also edit the README.md accordingly...

**Comment by martinezjavier** [comment]
@AndreasFuchsSIT got it, thanks a lot for the clarification.

File: test/integration/asymmetric-encrypt-decrypt.int.c [hunk 1]
```diff
@@ -140,6 +141,17 @@ test_invoke (TSS2_SYS_CONTEXT *sapi_context)
     LOG_INFO("Decrypt successed.");
 
     LOG_INFO("Asymmetric Encryption and Decryption Test Passed!");
+
+    rc = Tss2_Sys_FlushContext(sapi_context, sym_handle);
+    if (rc != TSS2_RC_SUCCESS) {
+        LOG_ERROR("Tss2_Sys_FlushContext failed with 0x%"PRIx32, rc);
+        return 99;
```
Comment (tstruk) [inline test/integration/asymmetric-encrypt-decrypt.int.c]: Could we add some comments about the 99 & 77 return codes?

Comment (AndreasFuchsSIT) [reply test/integration/asymmetric-encrypt-decrypt.int.c]: done

**Comment by AndreasFuchsSIT** [comment]
@tstruk updated...

**Pull request merged**